src/finn/custom_op/fpgadataflow/rtl/streamingfifo_rtl.py

```diff
@@ -95,12 +95,13 @@ def generate_hdl(self, model, fpgapart, clk):
         code_gen_dict["$TOP_MODULE_NAME$"] = topname
         # make instream width a multiple of 8 for axi interface
         in_width = self.get_instream_width_padded()
-        count_width = int(self.get_nodeattr("depth") - 1).bit_length()
+        count_width = int(self.get_nodeattr("depth") + 1).bit_length()
```

Comment:
No more `+1` needed. `bit_length()` already performs the correct computation:
```
>>> int(16).bit_length()
5
>>> int(15).bit_length()
4
```
The former code was really only off by one.

Reply:
Updated with latest commit to `count_width = int(self.get_nodeattr("depth")).bit_length()`